src_kt/java/arcs/util/Guard.kt

```diff
@@ -0,0 +1,85 @@
+package arcs.arcs.util
+
+import kotlinx.coroutines.sync.Mutex
```

Comment:
Is Mutex actually available in JS and WASM? It lives in common/ but the import of kotlin.jvm was worrying, I guess it's just the annotations: https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-core/common/src/sync/Mutex.kt

Reply:
It compiles at least....